op-chain-ops/cmd/celo-migrate/db.go

```diff
@@ -100,18 +101,22 @@ func removeBlocks(ldb ethdb.Database, numberHashes []*rawdb.NumberHash) error {
 	return nil
 }
 
-func getHeadHeader(dbpath string) (*types.Header, error) {
+func getHeadHeader(dbpath string) (headHeader *types.Header, err error) {
 	db, err := openDBWithoutFreezer(dbpath, true)
 	if err != nil {
 		return nil, fmt.Errorf("failed to open database at %q err: %w", dbpath, err)
 	}
-	defer db.Close()
+	defer func() {
+		if tempErr := db.Close(); tempErr != nil && err == nil {
+			err = fmt.Errorf("failed to close database: %w", tempErr)
+		}
+	}()
```

Comment:
I think you could use `errors.Join` here as well
```suggestion
	defer func() {
		err = errors.Join(err, db.Close())
	}()
```